airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -440,7 +453,35 @@ def create_job_yaml(self, context):
 
         return unique_job_name, yaml.safe_dump(kub_job_dict)
 
-    def execute(self, context):
+    @provide_session
+    def execute(self, context, session=None):
+
+        if self.die_if_duplicate:
+
+            current_task_instance = TaskInstance(self, context['execution_date'])
+            current_task_instance.refresh_from_db(include_queue_time=True)
+            TI = TaskInstance
```

Comment:
`TI` is just an alias of the `TaskInstance` class, right? I'm not sure I get why there is an alias here.

Reply:
We talked about this offline, but for posterity: this is just the pattern airflow uses for these types of queries.  Unsure if it's a sqlalchemy pattern or an airflow pattern, but either way, doing this to conform.